### PR TITLE
Create Mode to Replay Log in Order

### PR DESCRIPTION
**Pull request by ashmrtn**
Create a mode in CrashMonkey (that you can turn off with a flag) that will replay the logged workload in the order it was recorded. It will stop at each checkpoint in the workload to run `fsck` and user data tests. Update README with how to run this test.

### CODE REVIEW

File: code/harness/Tester.cpp [hunk 1]
```diff
@@ -529,102 +643,141 @@ int Tester::test_check_random_permutations(const int num_rounds,
...
-        test_info.fs_test.fsck_result += tmp;
-      }
+
+    // Required since this is a nested loop and `break` in the inner loop would
+    // just exit that loop, but still run the tests below (which we should skip
+    // if we don't have a Checkpoint?).
+    //
+    // TODO(ashmrtn): See if we should test that the entire log replay
+    // (i.e. when we reach the end of the log. If so, delete the below check).
+    if (log_iter == log_data.end()) {
```
Comment (ashmrtn) [inline code/harness/Tester.cpp]: Should we check the file system is consistent when we reach the end of the log, regardless of whether or not it's a checkpoint?

Comment (jayashreemohan29) [reply code/harness/Tester.cpp]: Yes, that sounds like a reasonable thing to do, as we expect a consistent state when all the bios are replayed.

**Comment by vijay03** [comment]
@ashmrtn can you please merge this?

**Comment by ashmrtn** [comment]
Let me made a few modifications to run the tests again after all the disk operations have been replayed as noted in comments above

**Pull request merged**